### ISSUE TITLE
Unknown lint `arithmetic_side_effects` (`rustc` warning)

### ISSUE THREAD

**Issue by Rudxain** [rust-lang/rust-clippy#9500]
### Description

I'm not sure if I should report this issue here or at `/rust-lang/rust/`. I was reading the docs, and found `clippy::arithmetic_side_effects`, set it to level `warn`, and `rustc` says that it can't recognize it. Why does this happen?

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by kraktus** [comment]
`arithmetic_side_effects` was added in [rust 1.64](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone), I assume this is the issue, but a reproducible example would be needed.

**Comment by Rudxain** [comment]
> `arithmetic_side_effects` was added in [rust 1.64](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone), I assume this is the issue, but a reproducible example would be needed.

Oh right, I'm gonna close the issue then. Sorry for the inconvenience.

BTW, I ran `rustup update && rustup upgrade`, but it didn't update the version. Am I doing something wrong?

**Comment by Rudxain** [comment]
I've updated to `rustc` 1.64, yet I still get the warning, even after `cargo clean`

**Comment by kraktus** [comment]
What is your clippy version?

**Comment by Rudxain** [comment]
Output of `cargo clippy -V`
```
clippy 0.1.64 (a55dd71 2022-09-19)
```

**Comment by KizzyCode** [comment]
Can confirm – same versions as OP, also running VSCode with `rust-analyzer`. I'm on macOS if it makes any difference.

**Comment by jmillikin** [comment]
I notice the lint was renamed from `arithmetic` to `arithmetic_side_effects` on September 9th (https://github.com/rust-lang/rust-clippy/commit/51d8b6c6643abb048987bd0befdeb6f46db0f6b5). It looks like Clippy v0.1.64 contains the old name:

```
// works
#![warn(clippy::arithmetic)]

// error: unknown lint: `clippy::arithmetic_side_effects`
#![warn(clippy::arithmetic_side_effects)]
```

Seems like a documentation bug that the name is updated immediately even if it stabilized under a different name.
